### PR TITLE
Manpage for lightning-listchannels created

### PR DESCRIPTION
**Pull request by runcrypto-michael**
Continuation of work on issue #1360
Resubmitting pull request after accidentally closing #2197.  This request contains changes suggested by @cdecker

### CODE REVIEW

File: doc/lightning-listchannels.7.txt [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+-----------
+The *listchannels* RPC command returns data on channels that are known to the
+node.  Because channels may be bidirectional, up to 2 objects will be returned
+for each channel (one for each direction).
+
+If no 'short_channel_id' is supplied, then data on all lightning channels known
+to this node, are returned.  These can be local channels or public channels
+broadcast on the gossip network.
+
+Supplying 'short_channel_id' will filter the results to only return data on a
```
Comment (niftynei) [inline doc/lightning-listchannels.7.txt]:
```suggestion
Supplying 'short_channel_id' will filter the results to only return data for 
```
grammar fixup

File: doc/lightning-listchannels.7.txt [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+- 'source' : The node providing entry to the channel, specifying the fees
+charged for using the channel in that direction.
+- 'destination' : The node providing the exit point for the channel.
+- 'short_channel_id' : The channel identifier.
+- 'public' : Boolean value, is publicly available.  Non-local channels will only
+ever have this value set to true.  Local channels are side-loaded by this node,
+rather than obtained through the gossip network, and so may have this value set
+to false.
+- 'satoshis' : The amount of funds available in the channel.
```
Comment (niftynei) [inline doc/lightning-listchannels.7.txt]:
```suggestion
- 'satoshis' : Funds available in the channel.
```
nit

File: doc/lightning-listchannels.7.txt [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+- 'message_flags' : Bitfield showing the presence of optional fields in the
+'channel_update' message (BOLT #7).
+- 'channel_flags' : Bitfield signalling values for optional fields defined in
+message_flags (BOLT #7).
```
Comment (niftynei) [inline doc/lightning-listchannels.7.txt]: `defined in message_flags` seems wrong for `channel_flags`

File: doc/lightning-listchannels.7.txt [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+- 'active' : Boolean value, is available for routing.  This is linked to the
+channel flags data, where if the second bit is set, signals a channels
+temporary unavailability (due to loss of connectivity) OR permanent
+unavailability where the channel has been closed but not settlement on-chain.	
+- 'last_update' : Unix timestamp showing when the last channel_update message
```
Comment (niftynei) [inline doc/lightning-listchannels.7.txt]: what's the resolution of `last_update`, seconds or milliseconds?

File: doc/lightning-listchannels.7.txt (hunk 3 above)
Comment (runcrypto-michael) [reply doc/lightning-listchannels.7.txt]: Yep, just re-read BOLT#7 and have confused reference to channel_update with channel_flags.  Will make this change now

File: doc/lightning-listchannels.7.txt (hunk 4 above)
Comment (runcrypto-michael) [reply doc/lightning-listchannels.7.txt]: Updated 'last_update' with resolution (seconds)

**Comment by rustyrussell** [comment]
Ack f93849f489f4c3f939c9834f4d1ae6a035be57ba

**Comment by rustyrussell** [comment]
Thanks, great contribution!

**Pull request merged**